pvlib/modelchain.py

```diff
@@ -1597,6 +1601,7 @@ def run_model(self, weather):
         pvlib.modelchain.ModelChain.run_model_from_poa
         pvlib.modelchain.ModelChain.run_model_from_effective_irradiance
         """
+        weather = _to_tuple(weather)
```

Comment:
Should move inside `prepare_inputs()`

Reply:
Ah yes I overlooked this earlier

Reply:
If it is moved inside `prepare_inputs()` then later in `ModelChain.run_model` we'll need to coerce type again for `ModelChain._run_from_effective_irrad` wouldn't we?

Reply:
Yes. Good catch. I guess we should just add redundant calls in `prepare_xxxx()`